readthedocs/api/v2/views/model_views.py

```diff
@@ -381,7 +377,27 @@ class RemoteRepositoryViewSet(viewsets.ReadOnlyModelViewSet):
     pagination_class = RemoteProjectPagination
 
     def get_queryset(self):
-        query = self.model.objects.api(self.request.user)
+        user = (
+            self.request.user
+            if self.request.user.is_authenticated
+            else None
+        )
+        query = self.model.objects.api(self.request.user).annotate(
+            admin=Case(
+                When(
```

Comment:
This query looks really good! I think it was a good idea to keep the compatibility. However, I'm a little worry about its performance. I never used `Case` and `When` and we will have a lot of relationships here.

We may need to put a TODO here with a message to checkout this after deploying it. 

Also, we can ask someone else what they think; but probably a TODO comment and a note in the GH project is fine for now.

Reply:
we can also use ``Exists`` with a subquery here. But I think this is quite efficient here. I'll add a ``# TODO`` and a note in the GH project. :)